src/main/scala/tv/teads/wiremock/extension/FreeMarkerRenderer.scala

```diff
@@ -128,7 +128,7 @@ class FreeMarkerRenderer extends ResponseDefinitionTransformer {
           val (filteredChildPath, filteredChildValue) = extractPathAndValueFromCondition(filteredChildCondition)
 
           arrayNode.elements().asScala
-            .find(findChildNode(_, filteredChildPath).textValue() == filteredChildValue)
```

Comment:
if the node is not a text node, this will return `null` thus prevent the filtering on number nodes for example.